Добавлен тест вкладок и тест оформления заказа на Chrome

Andrushka888 opened the pull request.

Andrushka888 committed:
Добавлен тест вкладок и тест оформления заказа на Chrome

mkouznetsov-git commented on src/test/java/TestDropdownListInChrome.java:
```diff
@@ -0,0 +1,43 @@
+    private WebDriver driver;
+
+    @Test
+    public void checkOpeningTabAboutСostAndTextWithinInChrome() {
```
Слово Cost начинается с русской буквы С. Лучше использовать только латиницу для наименований.

Andrushka888 replied:
Поправил

mkouznetsov-git commented on src/test/java/TestDropdownListInChrome.java:
```diff
@@ -0,0 +1,43 @@
+
+        assertEquals("Текст должен быть одинаковый", expectedText, actualText);
+
+    }
```
Не хватает тестов на другие 7 вопросов-ответов.

Andrushka888 replied:
Добавил данные и ещё 7 тестов

mkouznetsov-git replied:
Обрати внимание, локаторы вопросов-ответов различаются только одной цифрой. Можно их собирать из базовой строки плюс цифра, чтобы не делать 8 одинаковых переменных.
Также надо поступить и с тестами -- у тебя получилось 8 одинаковых тестов, логика проверки абсолютно идентичная. Значит, надо применить параметризацию. Параметризуй вопросы-ответы и будет один тестовый метод и 8 параметров.

mkouznetsov-git commented on src/test/java/TestDropdownListInChrome.java:
```diff
@@ -0,0 +1,43 @@
+    public void checkOpeningTabAboutСostAndTextWithinInChrome() {
+
+        driver = new ChromeDriver();
+        driver.get("https://qa-scooter.praktikum-services.ru/");
```
Подготовительные процедуры выноси из тестового метода в Before или в другое место. В тестовом методе надо сконцентрироваться только на стратегии проверки.

Andrushka888 replied:
Поправил

mkouznetsov-git commented on src/test/java/TestDropdownListInChrome.java:
```diff
@@ -0,0 +1,43 @@
+
+
+        objMainPage.clickCookieButton();
+        objMainPage.clickAboutСostButton();
```
Тут тоже русская буква С в слове Cost

Andrushka888 replied:
Поправил

mkouznetsov-git commented on src/test/java/TestOrderInChrome.java:
```diff
@@ -0,0 +1,65 @@
+    private WebDriver driver;
+
+    @Test
+    public void checkClickOrderUpButtonAndFillOrderAndCheckSuccessfulOrderWindowInChrome() {
```
Очень длинный метод. Разбей его на логические части и раздели на вспомогательные методы. Например, fillFormOnFirstPage() и т.п.
Тестовые данные лучше параметризовать.
Также здесь не хватает проверки другой кнопки заказа (их две). Эти кнопки (нижняя/верхняя) также добавь параметром к тесту и оставь один тестовый метод.

Andrushka888 replied:
Доработано. Тестовый метод разбит на логические части (По обеим вкладкам заказа), добавлена параметризация для тестовых данных, скорректирован метод нажатия на кнопки "Заказать", чтоб можно было указывать кнопку в тестовых данных.

Спасибо!

Andrushka888 committed:
Исправлены замечания.
Доработан тест заказа (Разбиение метода, параметризация, кнопки "Заказать")

mkouznetsov-git commented on src/main/java/org/example/MainPageScooter.java:
```diff
@@ -1,4 +1,220 @@
+    }
+
 }
+
```
Лишние пусты строчки и комментарии, не влияющие на понимание и/или работу кода следует безжалостно вырезать. )
Грамотное название классов, методов и переменных сам себя объясняет, это хороший стиль -- когда программа написана так, что читая её сразу быстро понимаешь, что там происходит, и не нужны дополнительные комментарии.

Andrushka888 replied:
Принято, поправил. 
В TestDropdownListInChrome добавлена параметризация, а из класса MainPageScooter убраны лишние локаторы

mkouznetsov-git commented on src/test/java/TestDropdownListInChrome.java:
```diff
@@ -0,0 +1,173 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+import static org.hamcrest.MatcherAssert.assertThat;
```
Этот импорт не используется

Andrushka888 replied:
Удалил все серые неиспользуемые импорты

mkouznetsov-git commented on src/test/java/TestOrderInChrome.java:
```diff
@@ -0,0 +1,127 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
```
Не используемый импорт

Andrushka888 replied:
Удалил все серые неиспользуемые импорты

mkouznetsov-git commented on src/test/java/TestOrderInChrome.java:
```diff
@@ -0,0 +1,127 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+
+import static org.hamcrest.MatcherAssert.assertThat;
```
Не используемый импорт

Andrushka888 replied:
Удалил все серые неиспользуемые импорты

mkouznetsov-git commented on src/test/java/TestOrderInChrome.java:
```diff
@@ -0,0 +1,127 @@
+       }
+    }
+
+//         public void checkClickOrderUpButtonAndFillOrderAndCheckSuccessfulOrderWindowInChrome() {
```
Комменты -- в топку )

Andrushka888 replied:
Убрал все лишние комменты. Это были пробы пера))

Andrushka888 committed:
Исправлены замечания.
Убраны лишние комментарии и пустые строки. Переработан TestDropdownListInChrome, добавлена параметризация и убраны лишние локаторы
